### PR TITLE
Command refactor

### PR DESCRIPTION
**Pull request by JuhaJGamer**
This is a pretty big thing I did.

Overhauled commands by separating UI logic from business logic
Simultaneously changed authorization model from prohibiting a command to prohibiting the action (business logic side authorization)

### CODE REVIEW

File: src/bot.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 import sys
 from aiohttp import web
 from accounting import LedgerServer, Authorization, RedditAccountId, DiscordAccountId, AccountId
-from commands import COMMANDS, list_commands_as_markdown, CommandException, assert_authorized, process_command
```
Comment (JuhaJGamer) [inline src/bot.py:12]: I do hope none of these show up in this file as they really shouldn't because otherwise you might be in trouble with running this

**Comment by JuhaJGamer** [comment]
ohh yeah i might have erased some two motnhs of command.py work














well except the stuff which was bug fixes whci hwas definitely fixed

**Comment by gamingdiamond982** [comment]
wow this is a neat solution

File: src/accounting.py [hunk 2]
```diff
@@ -127,12 +128,18 @@ def unwrap_proxies(account_id: AccountId) -> AccountId:
         return account_id
 
 
+@total_ordering
```
Comment (jonathanvdc) [inline src/accounting.py:131]: Making this class totally ordered is a nice QOL improvement. Thanks!

File: src/cli.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python3
```
Comment (jonathanvdc) [inline src/cli.py:1]: Love the CLI!

File: src/bot_commands.py [hunk 4]
```diff
@@ -0,0 +1,650 @@
...
+    commands.authorize(author, account, level, server)
+    return "Authorized"
+
+
+_add_command(
+    'authorize',
+    {
+        'account': (parse_account_id, "Account to authorize"),
+        'level': (
+            lambda s: {a.name.lower(): a for a in Authorization}[s.lower()],
```
Comment (jonathanvdc) [inline src/bot_commands.py:622]: Clever use of a dictionary comprehension!

**Review by jonathanvdc — APPROVED** [review]
These changes are pretty big so I can't say that I understand the full impact of every line in this PR, but I've read through the code in this PR and I think it's stunningly beautiful. Everything is so much neater now! Thank you so much for this fantastic refactor! It's clear that you put in a *lot* of work and I really appreciate that. This PR will allow us to grow the bot to meet future challenges.

P.S.: Sorry for the wait. I meant to review this PR earlier but life intervened.

**Comment by jonathanvdc** [comment]
Oh, this kinda broke the unit tests. That's my bad really. I should've set up CI to run on PRs (did that now).

**EDIT:** I fixed the unit tests so no worries.

**Pull request merged**